include/drivers/hwinfo.h

```diff
@@ -54,6 +66,30 @@ __syscall ssize_t hwinfo_get_device_id(uint8_t *buffer, size_t length);
 
 ssize_t z_impl_hwinfo_get_device_id(uint8_t *buffer, size_t length);
 
+/**
+ * @brief      Retrieve cause of device reset.
+ *
+ * @param      cause  OR'd flags TBD
+ *
+ * @retval zero if successful.
+ * @retval -ENOTSUP if there is no implementation for the particular device.
+ * @retval any negative value on driver specific errors.
+ */
+__syscall int hwinfo_get_reset_cause(u32_t *cause);
+
+int z_impl_hwinfo_get_reset_cause(u32_t *cause);
+
+/**
+ * @brief      Clear cause of device reset.
+ *
+ * @retval zero if successful.
+ * @retval -ENOTSUP if there is no implementation for the particular device.
+ * @retval any negative value on driver specific errors.
+ */
+__syscall int hwinfo_clear_reset_cause(void);
+
```

Comment:
@galak proposes an additional call that returns the bitmask of supported reasons in a particular MCU.